fuse/utils/ndict.py

```diff
@@ -162,84 +122,142 @@ def items(self) -> dict_items:
     def merge(self, other: dict) -> NDict:
         """
         inplace merge between self and other.
+
+        :param other: a dictionary
         """
-        other_flat = NDict(other).flatten()
-        for k, v in other_flat.items():
+        for k, v in other.items():
             self[k] = v
 
-        return
+        return self
 
     def __getitem__(self, key: str) -> Any:
         """
-        traverses the nested dict by the path extracted from spliting the key on '.', if key not found,
-        optionally shows the possible closest options
+        "traverses" the nested dict by the path extracted from splitting the key on '.'.
+        if key not found, shows the possible closest options.
+        if a prefix is given (the key exists but it is not a leaf), returns the corresponding sub dictionary
+
         :param key: dot delimited keypath into the nested dict
         """
-        nested_key = key.split(".")
-        if not nested_key[0] in self._stored:
+        # the key is a full key for a value
+        if key in self._stored:
+            return self._stored[key]
+
+        # the key is a prefix for other value(s)
+        elif self.is_prefix(key):  # TODO can be more optimized. we pass here once and in the "get_sub_dict" once again
```

Comment:
Please see `TODO` comment.

In my opinion is it enough to leave it like that for now (readability & code reuse VS optimization)

Reply:
remove is_prefix.
Instead return None is get_subdict